### DISCUSSION TITLE
train_text_to_image_sdxl training effect is very poor

### DISCUSSION THREAD

**Discussion by wzhiyuan2016** [huggingface/diffusers#9701] [Q&A]
I use DeepSpeed for training: train_text_to_image_sdxl.py 
1.The data volume is 231 pieces
2. deepspeed json
![企业微信截图_17291359065532](https://github.com/user-attachments/assets/f82ad033-d786-4fe4-9264-3b6236304170)
3.Training Script
![企业微信截图_17291362274700](https://github.com/user-attachments/assets/ae5a6207-dbc8-4dde-b5d7-dcdaa0ac2783)
4.After training, use the training prompt words again，The generated effect is as follows:
![企业微信截图_17291363542986](https://github.com/user-attachments/assets/004d3e51-de2e-453b-864a-803794659d2c)

May I ask everyone, what is the reason for the poor generation effect？

**Comment by a-r-r-o-w** [comment]
cc @sayakpaul

**Comment by sayakpaul** [comment]
This should be a discussion as there are far too many things that are unknown here. 

General comments:

* How do the samples look like in the dataset? 
* What is the resolution of the samples in the dataset? 
* Have you tried overfitting a single image and checked the results?

**Reply by wzhiyuan2016** [reply]
> This should be a discussion as there are far too many things that are unknown here.
> 
> General comments:
> 
> * How do the samples look like in the dataset?
> * What is the resolution of the samples in the dataset?
> * Have you tried overfitting a single image and checked the results?

hi 
* How do the samples look like in the dataset?
The data is similar to:
![企业微信截图_17326995273503](https://github.com/user-attachments/assets/fc713f6e-c75c-47c5-a2ad-8381daade85f)

![企业微信截图_17326995418545](https://github.com/user-attachments/assets/188b9a4e-17c3-4752-ad5c-6dd079b3ca75)
* What is the resolution of the samples in the dataset?
There are 213 pieces of training data, all of which are 1080p images, and the input for training is 1024 * 1024

* Have you tried overfitting a single image and checked the results?
Train using one image as shown in the following figure:
![image](https://github.com/user-attachments/assets/f5e57087-0d10-4201-a9e0-17798e8707cb)

"text": "Construction workers, Scaffolding, Building materials, Safety harness, Safety helmets, Yellow railings, Metal stairs, Concrete floor, Red and white safety tape, Tools and equipment"

The loss of training is as follows:
![1](https://github.com/user-attachments/assets/beb7d86a-b3ce-4922-8c4f-0dc2af66f52e)
![2](https://github.com/user-attachments/assets/1e0777b2-ffb7-4324-aef7-8318c719daf2)
![3](https://github.com/user-attachments/assets/861ea229-43fb-4e86-a62e-61e6efaa4732)
![4](https://github.com/user-attachments/assets/17c8e9f1-e29e-49fc-b7a7-b58ebd7c4f79)

inference script
![image](https://github.com/user-attachments/assets/d28f06a6-93d5-414b-9478-46f0f81cb300)

generate res:
![image](https://github.com/user-attachments/assets/70c9de53-a525-415b-9ae6-1488fb043e42)

**Reply by wzhiyuan2016** [reply]
@sayakpaul

**Comment by sayakpaul** [comment]
Does it work as expected without DeepSpeed? Have you tried varying the learning rate?

**Reply by wzhiyuan2016** [reply]
When I train train_text_to_image_stxl. py[Not applicable to Deepspeed], the parameter settings are as follows:[]
![image](https://github.com/user-attachments/assets/2d32ad00-70a3-426e-8dc9-e0808495ffb4)
![image](https://github.com/user-attachments/assets/e045bb01-c266-4edc-86fc-9cfa7406c3d6)
![image](https://github.com/user-attachments/assets/6514e981-0bbd-4fb2-a86b-594063d34390)
![image](https://github.com/user-attachments/assets/30cd1323-e431-479a-be02-5dfd624151aa)
The single card only has 24GB, and when setting BF16 for training, it reported CUDA out of memory
